build.gradle

```diff
@@ -36,41 +46,29 @@ sourceSets {
 }
 
 dependencies {
-	minecraft "com.mojang:minecraft:1.16.4"
-	mappings "net.fabricmc:yarn:1.16.4+build.9:v2"
-
 	// fabric-loader dependencies
-	implementation "org.ow2.asm:asm:${project.asm_version}"
-	implementation "org.ow2.asm:asm-analysis:${project.asm_version}"
-	implementation "org.ow2.asm:asm-commons:${project.asm_version}"
-	implementation "org.ow2.asm:asm-tree:${project.asm_version}"
-	implementation "org.ow2.asm:asm-util:${project.asm_version}"
-
-	// Required for mixin annotation processor
-	annotationProcessor "org.ow2.asm:asm:${project.asm_version}"
-	annotationProcessor "org.ow2.asm:asm-analysis:${project.asm_version}"
-	annotationProcessor "org.ow2.asm:asm-commons:${project.asm_version}"
-	annotationProcessor "org.ow2.asm:asm-tree:${project.asm_version}"
-	annotationProcessor "org.ow2.asm:asm-util:${project.asm_version}"
-
-	implementation('net.fabricmc:sponge-mixin:0.9.4+mixin.0.8.2') {
+	api "org.ow2.asm:asm:${project.asm_version}"
+	api "org.ow2.asm:asm-analysis:${project.asm_version}"
+	api "org.ow2.asm:asm-commons:${project.asm_version}"
+	api "org.ow2.asm:asm-tree:${project.asm_version}"
+	api "org.ow2.asm:asm-util:${project.asm_version}"
+
+	api('net.fabricmc:sponge-mixin:0.9.4+mixin.0.8.2') {
 		exclude module: 'launchwrapper'
 		exclude module: 'guava'
 	}
-	implementation 'net.fabricmc:tiny-mappings-parser:0.2.2.14'
-	implementation 'net.fabricmc:tiny-remapper:0.4.2'
-	implementation 'net.fabricmc:access-widener:1.0.0'
+	api 'net.fabricmc:tiny-mappings-parser:0.2.2.14'
+	api 'net.fabricmc:tiny-remapper:0.4.2'
+	api 'net.fabricmc:access-widener:1.0.0'
```

Comment:
Don't think we actually expose these in our APIs. Why expose them? They are also outdated, so worth an update if you do decide to make them public apis.

Reply:
These are only api so the minecraft project can compile against them if needed. We do not use whats in the maven in dev envs atm. I also wouldnt class this as exposing them as public apis. As far as im concerned this changes nothing to how it currently works.

Reply:
Hmm, might document this fact that these stuff are not guaranteed to be on a fixed version for loader somewhere, like on the wiki